plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerDelegate.java

```diff
@@ -65,12 +68,27 @@ public IStatus canModifyModules(IModule[] add, IModule[] remove) {
           if (status != null && !status.isOK()) {
             return status;
           }
+          status = hasAppEngineStandardFacet(module);
+          if (status != null && !status.isOK()) {
+            return status;
+          }
         }
       }
     }
     return Status.OK_STATUS;
   }
 
+  private IStatus hasAppEngineStandardFacet(IModule module) {
```

Comment:
These status objects should embed the module name.  You can add multiple modules and it may be just one module that fails.